pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -1,503 +1,8 @@
-            return self.checkpoint_io.save_checkpoint(checkpoint, filepath)
-
-    def remove_checkpoint(self, filepath: _PATH) -> None:
-        """Remove checkpoint filepath from the filesystem.
-
-        Args:
-            filepath: Path to checkpoint
-        """
-        if self.should_rank_save_checkpoint:
-            return self.checkpoint_io.remove_checkpoint(filepath)
-
-    @contextlib.contextmanager
-    def model_sharded_context(self) -> Generator:
-        """Provide hook to create modules in a distributed aware context. This is useful for when we'd like to
-        shard the model instantly, which is useful for extremely large models which can save memory and
-        initialization time.
-
-        Returns: Model parallel context.
-        """
-        yield
-
-    @abstractmethod
-    def teardown(self) -> None:
-        """This method is called to teardown the training process.
-
-        It is the right place to release memory and free other resources.
-        """
-        self._move_optimizer_state(torch.device("cpu"))
-
-    @classmethod
-    def register_plugins(cls, plugin_registry) -> None:
-        pass
-
-    @property
-    def should_rank_save_checkpoint(self) -> bool:
-        """Returns whether the checkpoint should be saved (rank based)"""
-        return self.is_global_zero
-
-    def on_train_start(self) -> None:
-        """Called when train begins."""
-        pass
-
-    def on_validation_start(self) -> None:
-        """Called when validation begins."""
-        pass
-
-    def on_test_start(self) -> None:
-        """Called when test begins."""
-        pass
-
-    def on_predict_start(self) -> None:
-        """Called when predict begins."""
-        pass
-
-    def on_train_end(self) -> None:
-        """Called when train ends."""
-        pass
-
-    def on_validation_end(self) -> None:
-        """Called when validation ends."""
-        pass
-
-    def on_test_end(self) -> None:
-        """Called when test end."""
-        pass
-
-    def on_predict_end(self):
-        """Called when predict ends."""
-        pass
-
-    def on_train_batch_start(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> None:
-        """Called in the training loop before anything happens for that batch."""
-        pass
-
-    def dispatch(self, trainer: "pl.Trainer") -> None:
-        """Hook to do something before the training/evaluation/prediction starts."""
-        self.precision_plugin.dispatch(trainer)
 
-    def post_dispatch(self, trainer: "pl.Trainer") -> None:
-        r"""
-        .. deprecated::
-            v1.6 This method has been deprecated in v1.6 and will be removed in v1.7. Use :meth:`teardown` instead.
+rank_zero_deprecation(
+    "TrainingTypePlugin was renamed to Strategy. Import `Strategy` from `pytorch_lightning.strategies.strategy`"
+)
 
-        Hook to do something after the training/evaluation/prediction finishes.
-        """
+TrainingTypePlugin = Strategy
```

Comment:
Here is what I'm trying so say: 
Everyone importing 
`from pytorch_lightning.plugins.x_plugin import XPlugin
`
will get a `XStrategy` and a deprecation message that they should change to

`from pytorch_lightning.strategies.x_strategy import XStrategy`